File: test_scrape.py
import requests

TARGET_URL = 'http://localhost:3000/api/claim'
TIKTOK_URL = 'https://www.tiktok.com/@bupati_xyz/video/1234567890'
TOKEN = 'test-token-dari-python'

print("--- Mulai Test Scrape (Python) ---")
print(f"> Mencoba request POST ke {TARGET_URL} tanpa header Referer")

# Data payload untuk request
payload = {
    "url": TIKTOK_URL,
    "token": TOKEN
}

# Request menggunakan requests (secara default tidak mengirim header Referer / Origin yang di-allow)
try:
    response = requests.post(TARGET_URL, json=payload)
    
    print(f"\nStatus Code: {response.status_code}")
    print(f"Response Body: {response.text}")
    
    if response.status_code == 403 or response.status_code == 500:
        print("\n✅ BERHASIL DIBLOKIR! Python bot ditolak aksesnya oleh server.")
    else:
        print("\n❌ GAGAL DIBLOKIR! Python bot berhasil mengakses API.")

except Exception as e:
    print(f"Terjadi error saat request: {e}")

File: test_scrape.js
const axios = require('axios');

const TARGET_URL = 'http://localhost:3000/api/claim';
const FAKE_TOKEN = 'test-token-12345';
const TIKTOK_URL = 'https://www.tiktok.com/@username/video/1234567890';

async function testScrape() {
  console.log('--- Mulai Test Scrape (Simulasi Bot/Scraper) ---');
  
  try {
    console.log(`\n> Mencoba menembak API ${TARGET_URL} tanpa Origin / Referer yang benar...`);
    
    // Request tanpa header Origin atau Referer yang valid (seperti layaknya scraper/bot biasa)
    const response = await axios.post(TARGET_URL, {
      url: TIKTOK_URL,
      token: FAKE_TOKEN
    }, {
      // Axios melempar error untuk status code 4xx dan 5xx secara default
      // Kita set validateStatus agar tidak throw exception untuk melihat response dari server
      validateStatus: function (status) {
        return status >= 200 && status <= 500;
      }
    });

    console.log(`Status Code Asli: ${response.status}`);
    console.log('Response Body:', response.data);

    if (response.status === 403 || response.status === 500) {
      console.log('✅ BERHASIL DIBLOKIR! Request bot/scraper ditolak karena restrictions (CORS/Referer).');
    } else {
      console.log('❌ GAGAL DIBLOKIR! Scraper bisa mengakses API.');
    }

  } catch (error) {
    console.error('Terjadi error saat request:', error.message);
  }
}

testScrape();
